llvm/lib/Transforms/Vectorize/VPlanConstruction.cpp

```diff
@@ -540,6 +540,9 @@ void VPlanTransforms::prepareForVectorization(
     if (auto *LatchExitVPB = MiddleVPBB->getSingleSuccessor())
```

Comment:
Independent: worth clarifying that not requiring a scalar epilog check means the scalar epilog is (always) required, i.e., case 1.